### DISCUSSION TITLE
How to build Gen2 images

### DISCUSSION THREAD

**Discussion by robertreems** [actions/virtual-environments#8430] [General questions]
Hi my customer is using the Automated image generation method for building our own images.

Since the change to managed images we've noticed that packer task now produces Ubuntu 22.04 Gen1 images where it previously generated Gen2 images. Looking into this issue we don't fully grasp why Gen2 images were created in the first place :-) But that is not the issue at hand here.

We'd like to know if there is any proper way to make sure packer will produce Gen2 images? Our downstream architecture is configured with Gen2 images. So we'd rather like to stick with that.
We already know that patching the 'ubuntu2204.pkr.hcl' and setting the SKU to '22_04-lts-gen2' works. But this approach doesn't feel right.

What are other possible methods in forcing packer to create a Gen2 image?

**Comment by vpolikarpov-akvelon** [comment]
Hey @robertreems. Changing the image SKU in the build process is typically done by patching the Packer template, and it's a straightforward approach. Could you please explain why you consider it a bad option?

**Reply by robertreems** [reply]
Hi [vpolikarpov-akvelon](https://github.com/vpolikarpov-akvelon),

Thanks for your response. 
We would like to keep the code from the repo as it is to minimize breakage of our own image build pipeline. We do a git clone, use the template within our own pipeline to build the images.

**Comment by robertreems** [comment]
We've chosen to patch the SKU in the packer template for now. But if there is any other way in configuring packer to produce Gen2 (HyperVgeneration V2) images it would be helpfull to us.
